lib/puppet/provider/docker_compose/ruby.rb

```diff
@@ -74,7 +74,7 @@ def get_image(service_name, compose_services)
 
   def create
     Puppet.info("Running compose project #{name}")
-    args = ['compose', compose_files, '-p', name, 'up', '-d', '--remove-orphans'].insert(3, resource[:options]).insert(5, resource[:up_args]).compact
+    args = ['compose', compose_files, '-p', name, 'up', '-d', '--remove-orphans'].insert(3, resource[:options]).insert(6, resource[:up_args]).compact
```

Comment:
The usage of `#insert` make it a bit hard to follow.  I think we could benefit from re-writing this as something like:

```ruby
args = ['compose', compose_files, '-p', name, resource[:options], 'up', resource[:up_args], '-d', '--remove-orphans'].flatten.compact
```
(assuming the index of the first insert is wrong because I don't think it is correct to insert `resource[:options]` between `-p` and `name`).

Also consistently using long options helps for readability: `-p` => `--project-name`, `-d` => `--detach`

Reply:
If y'all want to do it a different way that's certainly cool with me.  =)  I needed this fixed asap and submitted my fix without modifying the way things are done too much.

Reply:
(fwiw i agree -- i don't entirely understand why it was done the way it was)